src/main/java/net/openhft/chronicle/queue/util/NamedTableStoreLock.java

```diff
@@ -0,0 +1,71 @@
+package net.openhft.chronicle.queue.util;
+
+import net.openhft.chronicle.bytes.ref.BinaryLongReference;
+import net.openhft.chronicle.core.Jvm;
+import net.openhft.chronicle.queue.impl.TableStore;
+
+import java.io.IOException;
+import java.nio.channels.FileLock;
+import java.nio.channels.OverlappingFileLockException;
+
+import static java.lang.String.format;
+
+public class NamedTableStoreLock {
+
+    private static final int PID = Jvm.getProcessId();
+    private static final long UNLOCKED = Long.MIN_VALUE;
+
+    private final String lockName;
+    private final BinaryLongReference lockLongReference;
+
+    public NamedTableStoreLock(TableStore<?> tableStore, String lockName) {
+        this.lockName = lockName;
+        this.lockLongReference = (BinaryLongReference) tableStore.doWithExclusiveLock(ts -> ts.acquireValueFor(lockName, UNLOCKED));
+    }
+
+    public FileLock lock() {
+        return lock(lockLongReference, lockName);
+    }
+
+    public static FileLock lock(BinaryLongReference binaryLongReference, String lockName) {
+        FileLock lock;
+        try {
+            lock = binaryLongReference.lock(false);
+            if (lock != null) {
+                // This process acquired the replicator lock
```

Comment:
so this means the other process is dead?

Reply:
or no process has ever locked

Reply:
Either of those, you can't acquire the lock if its being held by another live process so a lock being returned means it was acquired.